Component access

atirutw opened the pull request:
I have no idea how Thisle's component system work, but I'll eventually figure this out.... probably.

Todo list:
- [ ] Component mapping
- [ ] Component access
- [x] Component listing

Bugs:
- [x] Corrupted last component

atirutw committed:
Stubs

atirutw committed:
Stubs for component mapper

atirutw committed:
Component mapper function prototypes

atirutw committed:
More consistent newlines

atirutw committed:
Implement component mapper functions

Do they work? Absolutely no idea!

atirutw committed:
Merge branch 'main' into component-access

atirut-w commented:
Small update nobody gaf: `exec_cmd()`, `get_cmd_status()` and `get_cmd_info()` confirmed working. One small step for my adventure into C and 6502 ig.

atirutw committed:
Codes to test component mapper

atirutw committed:
Store component count in a variable

atirutw committed:
Component struct and prototype for `list_components()`

I hope the UUID type is right...

atirutw committed:
Reduce component name length

atirutw committed:
Implement component listing

TODO: Clean up codes

atirutw committed:
Function for getting component count

atirutw committed:
Merge branch 'main' into component-access

atirutw committed:
Updated test code

atirutw committed:
Scrap some codes

atirutw committed:
Scrap a little more

atirutw committed:
Merge branch 'main' into component-access

atirutw committed:
Merge branch 'main' into component-access

atirutw committed:
Reimplement `get_component_count()`

atirutw committed:
Reimplement component listing

TODO: Fix corrupted last component

atirutw committed:
Last component no longer corrupted???

TODO: What the fuck

atirutw committed:
Rename struct `component` -> `componentinfo`

This would make more sense seeing as the things returned by the component list command are just basic info.

atirutw committed:
Component wrapper struct
